File: jejecrunch/programmers/javascript/42628.js
function solution(operations) {
    let answer = new Array()
    for (let i = 0; i < operations.length; i++) {
        const num = +operations[i].substring(2)
        switch (operations[i].substring(0, 1)) {
            case "I":
                answer.push(num)
                answer.sort((a, b) => a - b)
                break
            case "D":
                if (num === 1) answer.pop()
                else answer.shift()
                break
        }
    }

    if (answer.length === 0) return [0, 0]
    return [answer[answer.length - 1], answer[0]]
}

File: jejecrunch/programmers/42628.java
import java.util.*;
import java.util.regex.Pattern;

class Solution {
    static PriorityQueue<Integer> minQ = new PriorityQueue<>();
    static PriorityQueue<Integer> maxQ = new PriorityQueue<>(Collections.reverseOrder());

    public int[] solution(String[] operations) {
        int[] answer = new int[] { 0, 0 };

        for (String s : operations) {
            if (Pattern.matches("^[I\\s-?\\d]*$", s)) {
                minQ.add(Integer.parseInt(s.substring(2)));
                maxQ.add(Integer.parseInt(s.substring(2)));
            } else if (s.equals("D 1")) {
                if (minQ.size() < 1)
                    continue;
                minQ.remove(maxQ.poll());
            } else if (s.equals("D -1")) {
                if (minQ.size() < 1)
                    continue;
                maxQ.remove(minQ.poll());
            }
        }

        if (minQ.size() > 0) {
            answer[0] = maxQ.poll();
            answer[1] = minQ.poll();
        }

        return answer;
    }
}